### DISCUSSION TITLE
Number of database connections

### DISCUSSION THREAD

**Discussion by ank19** [eclipse-tractusx/tractusx-edc#1008] [Q&A] [answered]
Unfortunately, after the introduction of the new Policy-Monitor data source, we have reached the limit regarding the number of connections for PostgreSQL for our Azure Basic Tier plan.
 
- Is there any documentation about the requirements in this regard?
- Is there a way to influence the number of connections configuratively?

**Comment by florianrusch-zf** [comment]
Workaround: You can use the pgbouncer, which should be already available in Azure for your PostgreSQL instance.

**Reply by ank19** [reply, marked as answer]
Thanks, Florian, for your quick reply - that worked for us.

**Comment by ndr-brt** [comment]
maybe this could be also food for thoughts for the upstream implementation, currently, for every "sql context" you define in config with a `edc.datasource.<name>` a new pool is instantiated, with a default of max 8 total connection. 
If you keep the config separated for every store (asset, policy definition, ....) it will be easy to pass the 50 connections.

A little workaround, if you're using the same database/schema for every sql context, is to configure the `edc.datasource.default` so it will be used as a fallback, so then you can configure that pool independently

**Reply by ndr-brt** [reply]
somewhat related: https://github.com/eclipse-edc/Connector/issues/3811

**Reply by ank19** [reply]
Thanks for the hint, Andrea - we tried Florian's approach first, but if we have time, we will try the workaround you suggested, too.

**Comment by lgblaumeiser** [comment]
Closed as stale, please renew item if this is still an issue
